### PR TITLE
Move XmlSerializer CodeGen from SGen to S.P.Xml.

### PR DESCRIPTION
**Pull request by shmao**
The PR is moving the CodeGen code of XmlSerializer from SGen back to System.Private.Xml.

### CODE REVIEW

File: src/Microsoft.XmlSerializer.Generator/src/Sgen.cs [hunk 1]
```diff
@@ -263,7 +264,8 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
 
                     using (FileStream fs = File.Create(codePath))
                     {
-                        success = XmlSerializer.GenerateSerializer(serializableTypes, allMappings, fs);
+                        MethodInfo method = typeof(System.Xml.Serialization.XmlSerializer).GetMethod("GenerateSerializer", BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic);
```
Comment (huanwu) [inline src/Microsoft.XmlSerializer.Generator/src/Sgen.cs]: Can we check if method is null or not?

File: src/Microsoft.XmlSerializer.Generator/src/Sgen.cs [hunk 2]
```diff
@@ -263,7 +264,8 @@ private void GenerateFile(List<string> typeNames, string assemblyName, bool prox
...
+                        success = (bool)method.Invoke(null, new object[] { serializableTypes, allMappings, fs });
                     }
                 }
                 catch (UnauthorizedAccessException)
```
Comment (huanwu) [inline src/Microsoft.XmlSerializer.Generator/src/Sgen.cs]: Should we catch TargetInvocationException to catch the exception thrown by method invoked through reflection?

Comment (huanwu) [inline src/Microsoft.XmlSerializer.Generator/src/Sgen.cs]: Should we catch TargetInvocationException to catch the exception thrown by method invoked through reflection?

**Comment by shmao** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by shmao** [comment]
> System.Drawing.Common.Tests failed.

@dotnet-bot test Linux x64 Release Build

**Comment by shmao** [comment]
The PR passed the CI. I'm going to merge the changes into dev/sgen branch.

**Pull request closed without merging**